internal/pkg/service/common/etcdop/iterator/iterator.go

```diff
@@ -236,22 +236,23 @@ func (v *Iterator) moveToPage(resp *etcd.GetResponse) bool {
 	return true
 }
 
-func firstPageOp(prefix, end string, pageSize int, revision int64) op.GetManyOp {
-	return nextPageOp(prefix, end, pageSize, revision)
-}
-
-func nextPageOp(start, end string, pageSize int, revision int64) op.GetManyOp {
+func getPageOp(start, end string, pageSize int, revision int64, serializable bool) op.GetManyOp {
 	// Range options
 	opts := []etcd.OpOption{
 		etcd.WithFromKey(),
 		etcd.WithRange(end), // iterate to the end of the prefix
 		etcd.WithLimit(int64(pageSize)),
-		etcd.WithSort(etcd.SortByKey, etcd.SortAscend),
+		// etcd sorts results by default by the key (this is how the keys are physically stored).
+		// This sorting does not require loading all records into memory.
+		// The sort option is not needed.
+		// etcd.WithSort(etcd.SortByKey, etcd.SortAscend),
 	}
 
 	// Ensure atomicity
 	if revision > 0 {
 		opts = append(opts, etcd.WithRev(revision), etcd.WithSerializable())
+	} else if serializable {
+		opts = append(opts, etcd.WithSerializable())
 	}
 
```

Comment:
Pridana moznost nastavit `iterator` serializable read, by default je pouzity `linearizable`.

`linearizable` je pomalsia moznost, pri kazdom nacitani stranky sa skontroluje, ci > 1/2 clustra ma rovnaku verziu, ako node z ktorej citame.

`serializable` to nerobi, mozes tak nacitat o nieco starsie hodnoty, ale niekedy to nie je problem. `records` v etcd sa uz vramci uploadu nemenia ani nedoplnaju, takze mozeme pouzit tuto rychlejsiu moznost.